add location of the phone and overlay on the map

stanjouven opened the pull request:
- location in real time of the phone 
- overlay of the event
- go and see the schedule of an event when clicking on the marker
- customize markers with images

Commit:
add the current location feature and add a button to go to the current location of the phone

Commit:
modify the EventMapFragment so that it can handle the overlay feature

Commit:
add a method for the Zone class

Commit:
add a method for the Zone class

Commit:
class that connect the Zone class to the repository

Commit:
trying to fix code climate

Fnux reviewed (dismissed):
I don't know what this PR is about: the title doesn't seem to match the changes, there is no description or linked issue.

stanjouven committed:
merge master into fb-createMap to have fb-createMap branch up to date with master

stanjouven committed:
add an instance of ZoneModel in EventShowcaseActivity to add Zone class into the model

stanjouven committed:
add a ViewModel for ZoneModel into EventShowcaseModule

stanjouven committed:
delete the code that put positions manually for the overlay but instead taking all the positions from firebase

stanjouven committed:
add a test into SpotTest to test the getSpotType method

stanjouven committed:
add few tests to test Zone class

stanjouven committed:
Merge branch 'master' into fb-createMap

zbirobin committed:
change the API of Travis to 21 in order to make Travis display the map

zbirobin committed:
Revert "change the API of Travis to 21 in order to make Travis display the map"

This reverts commit 6d0f89f5e6aab8e7697df9d7ff57664482ca46ab.

stanjouven committed:
Added dimensions for custom marker

stanjouven committed:
Add custom cluster

custom cluster with images, and link the description of a marker with
his schedule (click on the description redirects to the corresponding
schedule)

stanjouven committed:
Add method to set tab when getting to ScheduleParentFragment

stanjouven committed:
Adapted Spot class to custom clusterer

stanjouven committed:
Merge branches 'fb-createMap' and 'master' of https://github.com/Susanfe/sdp-event-management into fb-createMap

# Conflicts:
#	app/src/main/java/ch/epfl/sweng/eventmanager/repository/EventRepository.java
#	app/src/main/java/ch/epfl/sweng/eventmanager/ui/eventShowcase/EventShowcaseActivity.java
#	app/src/main/res/values/dimens.xml

stanjouven committed:
Create multidrawable class for display few images in a cluster

zbirobin committed:
Inject firebase datas for the images of the markers

zbirobin committed:
Clean the code

zbirobin committed:
Clean more code

stanjouven committed:
added Zone to the mock repository

stanjouven committed:
Merge branch 'master' into fb-createMap

zbirobin committed:
change bad variable name

zbirobin committed:
Merge remote-tracking branch 'origin/fb-createMap' into fb-createMap

stanjouven committed:
added the Bitmap instance for constructing Spots in the tests

stanjouven committed:
Merge remote-tracking branch 'origin/fb-createMap' into fb-createMap

zbirobin committed:
Merge remote-tracking branch 'origin/fb-createMap' into fb-createMap

zbirobin committed:
Correct bug when clicking on schedule button

zbirobin committed:
Correct test

Luminilion committed:
Merge branch 'master' into fb-createMap

zbirobin committed:
Add a mocked repository for spots

zbirobin committed:
Add tests for tests for spotsModel

zbirobin committed:
delete useless variable

zbirobin committed:
Add a mocked repository for zone

zbirobin committed:
Add tests for zoneModel

zbirobin committed:
first part of merging with master (still compiler errors)

zyuiop committed:
Fix compilation errors

zyuiop committed:
Inject ViewModelFactory in the EventMapFragment

zbirobin committed:
finish merging with master and resolve bugs (still detects an error, but compiles)

zbirobin committed:
Merge remote-tracking branch 'origin/fb-createMap-merging-master' into fb-createMap-merging-master

zbirobin committed:
delete duplicate code

zbirobin committed:
correct the eventMapFragmentTest

zbirobin committed:
delete a duplicate mapFragmentTest

Fnux committed:
Merge branch 'master' into fb-createMap

stanjouven committed:
Merge branch 'master' into fb-createMap

stanjouven committed:
change the string returned by the getSnippet method -> The click to see the schedule text is handle in EventMapFragment

stanjouven committed:
correct code so that when clicking on the info window on map it directly go to this particular schedule

stanjouven committed:
change the appearance of the info window when clicking on a cluster -> to have separate lines in the info window

stanjouven committed:
Merge remote-tracking branch 'origin/fb-createMap' into fb-createMap

stanjouven committed:
xml code for the info window when clicking on a cluster on the map

zbirobin committed:
Merge branch 'master' into fb-createMap

zbirobin committed:
add a move that shows the different spots when clicking on a cluster

zbirobin committed:
decrease the ZOOMLEVEL

zbirobin committed:
Merge remote-tracking branch 'origin/fb-createMap' into fb-createMap

zbirobin committed:
change the travis file to make test the map

stanjouven committed:
Delete a textView that was for test the map but no more useful and delete some text when clicking to localisation button

stanjouven committed:
added test for Map

stanjouven committed:
Merge branches 'fb-createMap' and 'master' of https://github.com/Susanfe/sdp-event-management into fb-createMap

# Conflicts:
#	app/src/androidTest/java/ch/epfl/sweng/eventmanager/ui/event/interaction/fragments/EventMapFragmentTest.java
#	app/src/main/java/ch/epfl/sweng/eventmanager/ui/event/interaction/fragments/EventMapFragment.java
#	app/src/main/res/layout/fragment_event_map.xml
#	app/src/main/res/values/strings.xml

stanjouven committed:
Refactored scopes and added TODOs

stanjouven committed:
fix a problem in EventMapFragment

zbirobin committed:
refactor errors

zbirobin committed:
useless change to push again in order to have the diff coverage
